modules/data.atmosphere/tests/testthat/test.met.process.R

```diff
@@ -0,0 +1,19 @@
+test_that("`met.process` able to call .download.raw.met.module based on met process stage params", {
+  input_met <- list(source = 'CRUNCEP', id = '1')
+
+  mockery::stub(met.process, 'PEcAn.DB::db.open', 1)
+  mockery::stub(met.process, 'PEcAn.DB::db.close', 1)
+  mockery::stub(met.process, 'PEcAn.DB::db.query', list(file_path = '/test/path', file_name = 'test'))
+  mockery::stub(met.process, 'read.register', list())
+  mockery::stub(met.process, 'PEcAn.DB::query.format.vars', list())
+  mockery::stub(met.process, 'PEcAn.DB::dbfile.check', list(id = 1))
+  mockery::stub(met.process, 'assign', 1)
+  mockery::stub(met.process, 'db.site.lat.lon', list(lat = 0, lon = 0))
+  mockery::stub(met.process, 'met.process.stage', list(download.raw = TRUE, met2cf = FALSE, standardize = FALSE, met2model = FALSE))
```

Comment:
This sure does highlight how messy the met.process call stack is! Thanks for piecing this all together.